File: pdlearning/src/web-apps/src/libraries/domain-api/src/learner/models/my-learning-package.model.ts
export interface IMyLearningPackageModel {
  id: string;
  userId: string;
  myLectureId?: string | undefined;
  myDigitalContentId?: string | undefined;
  type: MyLearningPackageType;
  state: string;
  createdDate: Date;
  createdBy: string;
  changedDate?: Date | undefined;
  changedBy: string;
}

export class MyLearningPackageModel implements IMyLearningPackageModel {
  public id: string;
  public userId: string;
  public myLectureId?: string;
  public myDigitalContentId?: string;
  public type: MyLearningPackageType;
  public state: string;
  public createdDate: Date;
  public createdBy: string;
  public changedDate?: Date;
  public changedBy: string;

  constructor(data?: IMyLearningPackageModel) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.userId = data.userId;
    this.myLectureId = data.myLectureId;
    this.myDigitalContentId = data.myDigitalContentId;
    this.type = data.type;
    this.state = data.state;
    this.createdDate = new Date(data.createdDate);
    this.createdBy = data.createdBy;
    this.changedDate = data.changedDate !== undefined ? new Date(data.changedDate) : undefined;
    this.changedBy = data.changedBy;
  }
}

export enum MyLearningPackageType {
  SCORM = 'SCORM'
}

File: pdlearning/src/microservices/learner-microservice/Api/Application/Models/MyLearningPackageModel.cs
using System;
using Microservice.Learner.Domain.Entities;
using Microservice.Learner.Domain.ValueObject;

namespace Microservice.Learner.Application.Models
{
    public class MyLearningPackageModel
    {
        public MyLearningPackageModel(MyLearningPackage myLearningPackage)
        {
            Id = myLearningPackage.Id;
            UserId = myLearningPackage.UserId;
            MyLectureId = myLearningPackage.MyLectureId;
            MyDigitalContentId = myLearningPackage.MyDigitalContentId;
            Type = myLearningPackage.Type;
            State = myLearningPackage.State;
            LessonStatus = myLearningPackage.LessonStatus;
            CompletionStatus = myLearningPackage.CompletionStatus;
            SuccessStatus = myLearningPackage.SuccessStatus;
            CreatedDate = myLearningPackage.CreatedDate;
            CreatedBy = myLearningPackage.CreatedBy;
            ChangedDate = myLearningPackage.ChangedDate;
            ChangedBy = myLearningPackage.ChangedBy;
            TimeSpan = myLearningPackage.TimeSpan;
        }

        public Guid Id { get; set; }

        public Guid UserId { get; set; }

        public Guid? MyLectureId { get; set; }

        public Guid? MyDigitalContentId { get; set; }

        public LearningPackageType Type { get; set; }

        public string State { get; set; }

        public string LessonStatus { get; set; }

        public string CompletionStatus { get; set; }

        public string SuccessStatus { get; set; }

        public DateTime CreatedDate { get; set; }

        public Guid CreatedBy { get; set; }

        public DateTime? ChangedDate { get; set; }

        public Guid? ChangedBy { get; set; }

        public int? TimeSpan { get; set; }
    }
}
